src/Project/Project.jl

```diff
@@ -0,0 +1,427 @@
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ --- End License
+=#
+
+#=
+    The Project is the controller in an MVC paradigm. It manages the model,
+    stored in the projectDictionary, and plotting data, and responds to enableNotifications
+    of changes.
+=#
+mutable struct Project
+    name::String
+    projectDirectory:: String
+    projectDictionary::Dict{String,Any}
+    plt::Any #For the plot
+    plotOptions::Int # = combinations of MODEL, GRID, MESH
+#
+#   For drawing
+#
+    outerBndryPoints       ::Array{Any}              # CHAIN
+    outerBndryNames        ::Array{String}
+    innerBoundaryPoints    ::Array{Any}              # Array of CHAINs
+    innerBoundaryNames     ::Array{Any}              # Array of CHAINs of names
+    innerBoundaryChainNames::Array{String}           # Array of the names of the CHAINs
+    refinementRegionPoints ::Array{Array{Float64,2}} # Array of Array of points
+    refinementRegionNames  ::Array{String}
+    refinementRegionLoc    ::Array{Array{Float64}}   # Center point of a refinement region
+    bounds                 ::Array{Float64}
+    userBounds             ::Array{Float64}
+    xGrid                  ::Array{Float64}
+    yGrid                  ::Array{Float64}
+    xMesh                  ::Array{Float64}
+    yMesh                  ::Array{Float64}
+    backgroundGridShouldUpdate::Bool
+    meshShouldUpdate        ::Bool
+end
+
+const defaultPlotPts  = 50
+const meshFileFormats = Set(["ISM", "ISM-V2", "ABAQUS"])
+const plotFileFormats = Set(["sem", "skeleton"])
+const smootherTypes   = Set(["LinearSpring", "LinearAndCrossbarSpring"])
+const statusValues    = Set(["ON", "OFF"])
+const refinementTypes = Set(["smooth", "sharp"])
+
+
+"""
+    openProject(fileName::String, folder::String)
+
+Open existing project described in the control File.
+
+    folder   = folder the control file is in
+    fileName = the name of the file
+"""
+function openProject(fileName::String, folder::String)
```

Comment:
This should just be called `Project` since it acts as a constructor.
```suggestion
function Project(fileName::String, folder::String)
```

Reply:
No. It opens a file for the project, and describes just what it does. If I want a new project, I call `newProject`. If I want to open an existing one (like above) I will call `openProject`. Need a verb and a noun.